PR #788 breaks build of mupen64plus-video-rice

The rice plugin cannot be build anymore against the current mupen64plus-core since c9def89c0a08bd95067cab7acd5b2fa59446f525:

    In file included from ../../src/liblinux/BMGDLL.h:48:0,
                     from ../../src/TextureFilters.cpp:44:
    ../../src/liblinux/tiffrw.h:33:28: error: redeclaration of 'NONE'
     enum TiffCompressionEnum { NONE, CCITTRLE, CCITTFAX3, CCITTFAX4, LZW, JPEG6,
                                ^
    In file included from ../../../mupen64plus-core/src/api/m64p_config.h:29:0,
                     from ../../src/Video.h:30,
                     from ../../src/RenderBase.h:26,
                     from ../../src/TextureFilters.cpp:23:
    ../../../mupen64plus-core/src/api/m64p_types.h:232:4: note: previous declaration 'm64p_rom_save_type NONE'
        NONE

Please never ever add some symbol or constant definition without project specific prefix to a global public include file.

This was done by #788 Rosalie241/settings which was merged by @richard42

--- a/src/api/m64p_types.h
+++ b/src/api/m64p_types.h
@@ -224,12 +224,12 @@ typedef enum
 
 typedef enum
 {
-   EEPROM_4KB = 0,
-   EEPROM_16KB,
-   SRAM,
-   FLASH_RAM,
-   CONTROLLER_PACK,
-   NONE
+    SAVETYPE_EEPROM_4KB = 0,
+    SAVETYPE_EEPROM_16KB,
+    SAVETYPE_SRAM,
+    SAVETYPE_FLASH_RAM,
+    SAVETYPE_CONTROLLER_PACK,
+    SAVETYPE_NONE
 } m64p_rom_save_type;
 
 typedef struct

--- a/src/device/cart/cart.c
+++ b/src/device/cart/cart.c
@@ -129,9 +129,9 @@ void init_cart(struct cart* cart,
     init_sram(&cart->sram,
         sram_storage, isram_storage);
 
-    if (ROM_SETTINGS.savetype == SRAM)
+    if (ROM_SETTINGS.savetype == SAVETYPE_SRAM)
         cart->use_flashram = -1;
-    else if (ROM_SETTINGS.savetype == FLASH_RAM)
+    else if (ROM_SETTINGS.savetype == SAVETYPE_FLASH_RAM)
         cart->use_flashram = 1;
     else
         cart->use_flashram = 0;

--- a/src/main/main.c
+++ b/src/main/main.c
@@ -1064,7 +1064,7 @@ static void open_eep_file(struct file_storage* fstorage)
     }
 
     /* Truncate to 4k bit if necessary */
-    if (ROM_SETTINGS.savetype != EEPROM_16KB) {
+    if (ROM_SETTINGS.savetype != SAVETYPE_EEPROM_16KB) {
         fstorage->size = 0x200;
     }
 }
@@ -1678,7 +1678,7 @@ m64p_error main_run(void)
                 vi_clock_from_tv_standard(ROM_PARAMS.systemtype), vi_expected_refresh_rate_from_tv_standard(ROM_PARAMS.systemtype),
                 NULL, &g_iclock_ctime_plus_delta,
                 g_rom_size,
-                (ROM_SETTINGS.savetype != EEPROM_16KB) ? JDT_EEPROM_4K : JDT_EEPROM_16K,
+                (ROM_SETTINGS.savetype != SAVETYPE_EEPROM_16KB) ? JDT_EEPROM_4K : JDT_EEPROM_16K,
                 &eep, &g_ifile_storage,
                 flashram_type,
                 &fla, &g_ifile_storage,

--- a/src/main/rom.c
+++ b/src/main/rom.c
@@ -193,7 +193,7 @@ m64p_error open_rom(const unsigned char* romimage, unsigned int size)
     {
         strcpy(ROM_SETTINGS.goodname, ROM_PARAMS.headername);
         strcat(ROM_SETTINGS.goodname, " (unknown rom)");
-        ROM_SETTINGS.savetype = NONE;
+        ROM_SETTINGS.savetype = SAVETYPE_NONE;
         ROM_SETTINGS.status = 0;
         ROM_SETTINGS.players = 4;
         ROM_SETTINGS.rumble = 1;
@@ -523,22 +523,22 @@ void romdatabase_open(void)
             else if(!strcmp(l.name, "SaveType"))
             {
                 if(!strcmp(l.value, "Eeprom 4KB")) {
-                    search->entry.savetype = EEPROM_4KB;
+                    search->entry.savetype = SAVETYPE_EEPROM_4KB;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else if(!strcmp(l.value, "Eeprom 16KB")) {
-                    search->entry.savetype = EEPROM_16KB;
+                    search->entry.savetype = SAVETYPE_EEPROM_16KB;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else if(!strcmp(l.value, "SRAM")) {
-                    search->entry.savetype = SRAM;
+                    search->entry.savetype = SAVETYPE_SRAM;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else if(!strcmp(l.value, "Flash RAM")) {
-                    search->entry.savetype = FLASH_RAM;
+                    search->entry.savetype = SAVETYPE_FLASH_RAM;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else if(!strcmp(l.value, "Controller Pack")) {
-                    search->entry.savetype = CONTROLLER_PACK;
+                    search->entry.savetype = SAVETYPE_CONTROLLER_PACK;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else if(!strcmp(l.value, "None")) {
-                    search->entry.savetype = NONE;
+                    search->entry.savetype = SAVETYPE_NONE;
                     search->entry.set_flags |= ROMDATABASE_ENTRY_SAVETYPE;
                 } else {
                     DebugMessage(M64MSG_WARNING, "ROM Database: Invalid save type on line %i", lineno);

